system/Pager/Pager.php

```diff
@@ -295,19 +302,30 @@ public function getFirstPage(string $group = 'default')
 	/**
 	 * Returns the URI for a specific page for the specified group.
 	 *
-	 * @param int    $page
-	 * @param string $group
-	 * @param bool   $returnObject
+	 * @param int|null $page
+	 * @param string   $group
+	 * @param bool     $returnObject
 	 *
-	 * @return string
+	 * @return string|\CodeIgniter\HTTP\URI
 	 */
 	public function getPageURI(int $page = null, string $group = 'default', $returnObject = false)
 	{
 		$this->ensureGroup($group);
 
 		$uri = $this->groups[$group]['uri'];
 
```

Comment:
Another subject, but related to this: Thinking of a way to do not re-encode the query string. Because this is breaking the original strings.

I believe that this can be solved here or in the https://github.com/bcit-ci/CodeIgniter4/blob/7f85f539cedeb90e3ea6b37781357006f50adedd/system/HTTP/URI.php#L739-L753

Otherwise it is necessary to set the `$_GET` before calling pagination as: `$_GET['search'] = urldecode($_GET['search']);`

Reply:
The latest code in the repo should have the encoding removed. Have you updated in the last couple of days?

Reply:
Yes. Try it on a blank page:

```php
$pager = \Config\Services::pager();
echo $pager->makeLinks($_GET['page'] ?? 0, 10, 200);
```

In the browser try something like: http://localhost:8080/?q=you+alice+to+the+little

Navigate by the links. Look that the URI continues changing:

Page 2: `http://localhost:8080/?q=you%2Balice%2Bto%2Bthe%2Blittle&page=2`

Page 3: `http://localhost:8080/?q=you%252Balice%252Bto%252Bthe%252Blittle&page=3`

Page 4: `http://localhost:8080/?q=you%25252Balice%25252Bto%25252Bthe%25252Blittle&page=4`

And continues re-encoding... I'm testing with PHP and nginx servers, and same results.